### PR TITLE
Mirroring C enums in Python for CMontecarlo tests.

### PR DESCRIPTION
**Pull request by karandesai-96**
There are certain parameters, such as `RPacket.status` and `StorageModel.cont_status` which are C enum datatypes. Enums were considered as `ctypes.c_uint` or `ctypes.c_int` in Python, and implemented in that manner.

This PR adds a new file, **enum.py** which sets up a mirror for C enums in python. Enums are not supported directly in `ctypes` so a Meta Class and corresponding Class were written.

Following were mirrored for Python tests and used:
- [X] TardisError
- [X] RPacketStatus
- [X] ContinuumProcessesStatus

This targets improvement in readability of tests.

### CODE REVIEW

**Comment by wkerzendorf** [comment]
That looks like an interesting addition and makes the code more readable - well done @karandesai-96

File: tardis/montecarlo/enum.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
+from ctypes import c_int
+
+
+class EnumerationType(type(c_int)):
```
Comment (wkerzendorf) [inline tardis/montecarlo/enum.py]: Normally, when we copy code from the internet we like to attribute it with the original source. In science this is an important part to understand the source of information. Can you add one?

File: tardis/montecarlo/enum.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        if isinstance(other, int):
+            return self.value == other
+
+        return type(self) == type(other) and self.value == other.value
+
+    def __repr__(self):
+        value = self.value
+        return "<%s.%s: %d>" % (self.__class__.__name__,
+                                self._reverse_map_.get(value, '(unknown)'),
+                                value)
```
Comment (yeganer) [inline tardis/montecarlo/enum.py]: You can directly write self.value here

**Comment by karandesai-96** [comment]
Thanks @wkerzendorf. I will work upon the suggestions received here and wrap up here with a couple of docstrings ;)

File: tardis/montecarlo/enum.py (hunk 1 above)
Comment (karandesai-96) [inline tardis/montecarlo/enum.py]: - [X] Added.

File: tardis/montecarlo/enum.py (hunk 2 above)
Comment (karandesai-96) [inline tardis/montecarlo/enum.py]: - [X] Done.

**Comment by wkerzendorf** [comment]
I'm signing off on this piece of code. It looks good (as far as I understand). @yeganer - you can merge it tomorrow if you are happy as well.

**Comment by yeganer** [comment]
Looks good to me, I'll merge.

**Comment by karandesai-96** [comment]
This modification is causing occasional segfaults, so it needs to be modified.

**Pull request merged**